Update pom.xml to deploy artifacts to Maven central

harawata opened the pull request:
Hello @basil ,

I was interested in #22 and read the discussion, but wasn't sure what the conclusion was (English is not my primary language).
So, I figured that it would be simpler to show you what is necessary.

This basically follows the Sonatype's instruction.
https://central.sonatype.org/publish/publish-maven/#performing-a-release-deployment

I already have an account on Sonatype and am publishing several artifacts.
With this modification, it should be possible for me to deploy the artifact to the Maven Central manually (no GitHub Actions needed).
You might have to vouch for me when/if they ask you, but that's pretty much everything you need to do AFAIK.

If this is not what you want, feel free to close this PR. I promise I won't be offended :)
If you have any question, please let me know.

### Testing done

As this change does not include any logical change, there is no new tests.

### Submitter checklist

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

basil reviewed (changes requested):
Since the last time I looked at this, https://github.com/jenkinsci/central-repository-pom was introduced. For consistency with other components in the Jenkins project, I think we should switch to that parent POM and its associated processes (including having Alex publish releases in the short term, which could be expanded to more people over time as needed).

harawata commented:
Thanks for the swift reply!
I have updated the PR (simply replaced the parent and added a couple of `<version>`s).
Please let me know if there need to be more adjustments.

basil commented on pom.xml:
```diff
@@ -2,9 +2,9 @@
-    <version>1.110</version>
+    <groupId>io.jenkins.tools</groupId>
+    <artifactId>central-repository-pom</artifactId>
+    <version>1.2</version>
```
```suggestion
    <version>1.3</version>
```

basil reviewed (commented):
Changes to the POM look correct to me.

@NotMyFault Are we authorized to publish new releases of `org.jvnet.mock-javamail:mock-javamail` to Maven Central? And if we are, are you interested and willing to perform the releases? If so, then I think we can move forward with this PR.

NotMyFault commented:
> Are we authorized to publish new releases of `org.jvnet.mock-javamail:mock-javamail` to Maven Central?

According to [OSSRH-2216](https://issues.sonatype.org/browse/OSSRH-2216), only Kohsuke can cut releases to this groupId.

If we want to distribute this artifact on the central repository, I could drive the effort to reach out to Kohsuke and onboard myself with release permissions including [everyone](https://github.com/jenkinsci/central-repository-pom/blob/653413b319d1345c39ab18f6b43ed3aa54d1f560/pom.xml#L21-L68) with preexisting release permission of common Jenkins components, if desired.
For the record, everyone defined in the developers block of components we publish to the central repository can cut releases.

Do you want to be added to this and `io.jenkins.tools` namespace too, Basil?

basil commented:
> If we want to distribute this artifact on the central repository, I could drive the effort to reach out to Kohsuke and onboard myself with release permissions including [everyone](https://github.com/jenkinsci/central-repository-pom/blob/653413b319d1345c39ab18f6b43ed3aa54d1f560/pom.xml#L21-L68) with preexisting release permission of common Jenkins components, if desired.

This sounds great to me!

> Do you want to be added to this and `io.jenkins.tools` namespace too, Basil?

Sure, good to have more active developers able to cut releases.

NotMyFault commented:
> Sure, good to have more active developers able to cut releases.

Agreed, do you have an account on the central repository already I can reference to? If not, you may need to [create](https://issues.sonatype.org/secure/Signup!default.jspa) one and let me know your username.

NotMyFault commented:
> Sure, good to have more active developers able to cut releases.

Agreed, do you have an account on the central repository already, @basil? If not, you may need to [create](https://issues.sonatype.org/secure/Signup!default.jspa) one and let me know your username so I can get you added.

basil commented:
> Agreed, do you have an account on the central repository already, @basil? If not, you may need to [create](https://issues.sonatype.org/secure/Signup!default.jspa) one and let me know your username so I can get you added.

Thank you @NotMyFault, I just created a [basilcrow](https://issues.sonatype.org/secure/ViewProfile.jspa?name=basilcrow) account.

basil commented:
@NotMyFault Are we still waiting on Kohsuke for this one?

NotMyFault commented:
> @NotMyFault Are we still waiting on Kohsuke for this one?

Yes, I have emailed @kohsuke, but got no response.

oytuntez commented:
Any updates? I'd love to use 2.2.

basil commented:
Hi @kohsuke, gentle ping about the OSSRH portion of this.
